test/e2e_node/critical_pod_test.go

```diff
@@ -190,9 +191,13 @@ func getTestPod(critical bool, name string, resources v1.ResourceRequirements, n
 		}
 		pod.Spec.PriorityClassName = scheduling.SystemNodeCritical
 
-		framework.ExpectEqual(kubelettypes.IsCriticalPod(pod), true, "pod should be a critical pod")
+		if !kubelettypes.IsCriticalPod(pod) {
+			framework.Failf("pod %q should be a critical pod", pod.Name)
```

Comment:
Glancing over this PR, is this the original behavior of a failed ExpectEqual was to fail? Your change introduces a failure if the match fails which may be unnecessary for deprecation?

Reply:
@kannon92 FYI: https://github.com/kubernetes/kubernetes/issues/105678

Reply:
`framework.ExpectEqual(..., true)` is hard to read, IMHO the if check is definitely an improvement. :+1: